app/main.js

```diff
@@ -91,6 +92,23 @@ window.initApp = () => {
   }
 };
 
+const detection = new UnsupportedBrowserDetection();
+const checkSupportedBrowser = () => {
+  if (detection.isSupported()) {
+    setTimeout(() => {
+      window.initApp();
+    });
+  }
+};
+const setOnLoadCheckSupportedBrowserListener = () => {
+  window.onload = () => {
+    checkSupportedBrowser();
+  };
+};
+document.readyState === 'complete' ? checkSupportedBrowser() : setOnLoadCheckSupportedBrowserListener();
```

Comment:
What's the point of this condition?

Reply:
The initial problem was that sometimes window.onload was called before initApp was defined. This way it will be called regardless of onload being called before or after its defined.

Reply:
I don't get it. It's literally the same file. You control when it's defined.

Reply:
Do You suggest to skip waiting/checking if everything is loaded and only check browser support before lunching appInit?

Reply:
Yeah, because there's no need to really wait for anything. I'm not even sure that we need the onload event here.

Reply:
In that case we wont need it at all